src/lib.rs

```diff
@@ -78,69 +80,73 @@ impl Agent {
 
 // ***** Income evolution *****
 
-pub fn income(agent: Agent, rate: f32) { // add rate to global parameters
-    let a = agent.age;
-    let i = agent.social_class.init_income;
-    let g = agent.social_class.growth;
-    let s = agent.social_class.start;
-    let r = agent.social_class.retire + 1;
-    if a < s {
-        0
-    } else if a < r {
-        i * g.pow(a - s) 
-    } else {
-        i * r
+impl Agent {
+    // TODO add rate to global parameters
+    pub fn income(&self, rate: f32) -> f32 {
+        let age = self.age;
+        let init_income = self.social_class.init_income;
+        let start = self.social_class.start;
+        let retire = self.social_class.retire + 1;
+        if age < start {
+            0.0
+        } else if age < retire {
+            init_income * self.social_class.growth.pow(age - start)
+        } else {
+            init_income * retire
+        }
     }
 }
 
 // ***** Behaviour functions *****
 
-fn income_in_wneigh(s_network: SNetwork, adress: Address, dist: f32) { // add walking distance to define walking neigh to global parameters
+fn income_in_wneigh(s_network: SNetwork, adress: Address, dist: f32) -> Vec<i32> { // add walking distance to define walking neigh to global parameters
```

Comment:
What will this method produce as output? A list of integers? Later on you do `let (j,l,s) = activity_in_tneigh(s_network, agent.house.adress, dist);` -- does it return exactly 3 things? You can say `-> (i32, i32, i32)` if so